ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs

```diff
@@ -180,6 +181,9 @@ data RunNodeArgs m addrNTN addrNTC blk (p2p :: Diffusion.P2P) = RunNodeArgs {
 
       -- | Network PeerSharing miniprotocol willingness flag
     , rnPeerSharing :: PeerSharing
+
+      -- | An event handler to trigger custom action when ledger events are emitted.
```

Comment:
Regarding the long-term plan. I think this data flow would be unnecessary if the the ChainDB instead stored the ledger events alongside each block. Then those events could simply be served alongside the (potentially immutable) blocks (and rollback messages!) by the local ChainSync client.

But that would of course require the additional on-disk database that the PR description posited.

Reply:
So I wonder of the effort should be spent there instead of on what currently seems to me to be a stop-gap measure.

Reply:
> what currently seems to me to be a stop-gap measure.

What do you mean?

Reply:
While a dedicated database alongside the `ImmutableDB` and friends would be desirable as it would remove the need for clients to track themselves the data, the proposed solution has the advantage of the simplicity and not adding an additional responsibility on top of an already pretty busy consensus. 
In practice, we thought a client that crashes or wants to catch up could simply reset the node before the desired point and let the synchronisation magic happens. Doing it manually is error prone but relatively straightforward, of course one would want to have a tool for that but this is orthogonal to streaming the events.